intelmq/bots/experts/aggregate/expert.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+"""
+Aggregate Expert
+
+SPDX-FileCopyrightText: 2021 Intelmq Team <intelmq-team@cert.at>
+SPDX-License-Identifier: AGPL-3.0-or-later
+"""
+from datetime import datetime, timedelta
+import time
+import json
+from intelmq.lib.bot import Bot
+from intelmq.lib.utils import parse_relative
+from intelmq.lib.mixins import CacheMixin
+
+
+class AggregateExpertBot(Bot, CacheMixin):
+    """Aggregation expert bot"""
+
+    fields: str = "classification.type, classification.identifier"
+    threshold: int = 10
+    redis_cache_db: int = 8
+    redis_cache_host: str = "127.0.0.1"  # TODO: could be ipaddress
+    redis_cache_password: str = None
+    redis_cache_port: int = 6379
+    timespan: str = "1 hour"
+
+    __timespan: int = 0
+    __next_cleanup: int = 0
+
+    def init(self):
+        self.__timespan = parse_relative(self.timespan)
+        self.fields = {k.strip() for k in self.fields.split(',')}
+        self.cleanup()
+
+    def cleanup(self):
+        if self.__next_cleanup <= time.time():
+            delta = timedelta(minutes=self.__timespan)
+            for key in self.cache_get_redis_instance().keys(pattern="aggregate.*"):
+                keys = self.cache_get_redis_instance().hgetall(key)
+                data = {y.decode('utf-8'): keys.get(y).decode('utf-8')
+                        for y in keys.keys()}
+
+                if datetime.now().isoformat() <= (datetime.strptime(data['s'], '%Y-%m-%dT%H:%M:%S.%f') + delta).isoformat():
```

Comment:
While comparing time as ISO-formatted strings is possible, is this conversion step necessary?

Reply:
Does this work?
```suggestion
                if datetime.now() <= (datetime.strptime(data['s'], '%Y-%m-%dT%H:%M:%S.%f') + delta):
```